src/driver.rs

```diff
@@ -0,0 +1,378 @@
+use crate::api::QldbSessionApi;
+use crate::rusoto_ext::*;
+use crate::transaction::{Transaction, TransactionAttempt, TransactionDisposition};
+use crate::{
+    pool::SessionPool, retry::default_retry_policy, retry::TransactionRetryPolicy, QldbError,
+};
+use runtime::Runtime;
+use rusoto_core::{
+    credential::{DefaultCredentialsProvider, ProvideAwsCredentials},
+    Client, HttpClient, Region, RusotoError,
+};
+use rusoto_qldb_session::*;
+use std::error::Error as StdError;
+use std::{cell::RefCell, future::Future};
+use tokio::{runtime, sync::mpsc::channel};
+
+/// A builder to help you customize a [`QldbDriver`].
+///
+/// In many cases, it is sufficient to use [`QldbDriver::new`] to build a driver out of a Rusoto client for a particular QLDB ledger. However, if you wish to customize the driver beyond the
+/// defaults, this builder is what you want.
+///
+/// Note that the following setters _must_ be called, else [`build`] will return an `Err`:
+/// - `ledger_name`
+/// - `client`
+///
+/// Usage example:
+/// ```no_run
+/// # use amazon_qldb_driver::QldbDriverBuilder;
+/// # use rusoto_core::region::Region;
+/// # use rusoto_qldb_session::QldbSessionClient;
+/// #
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let client = QldbSessionClient::new(Region::UsWest2);
+/// let driver = QldbDriverBuilder::new()
+///     .ledger_name("sample-ledger")
+///     .region(Region::UsEast1)
+///     .build()?;
+/// # Ok(())
+/// # }
+/// ```
+pub struct QldbDriverBuilder {
+    ledger_name: Option<String>,
+    credentials_provider: BoxedCredentialsProvider,
+    region: Option<Region>,
+    transaction_retry_policy: Box<dyn TransactionRetryPolicy>,
+    max_sessions: usize,
```

Comment:
For all the other drivers, we have changed to use `maxConcurrentTransactions`, since the concept of session are not exposed in drivers anymore.

Reply:
ack, will make the change

Reply:
https://github.com/awslabs/amazon-qldb-driver-rust/pull/13